### PR TITLE
Text to speech

### PR DESCRIPTION
**Pull request by jskeet**
I'm not proposing to merge this to master yet:

- The protos/yaml needed hand-editing
- The gRPC end-point doesn't appear to be enabled yet

This PR will merge into a new branch instead.

### CODE REVIEW

**Comment by chrisdunelm** [comment]
Shouldn't there be an appveyor build?

File: apis/apis.json [hunk 1]
```diff
@@ -635,6 +635,16 @@
     }
   },
 
+  {
+    "id": "Google.Cloud.TextToSpeech.V1",
+    "productName": "Google Cloud Text-to-Speech",
+    "productUrl": "https://cloud.google.com/text-to-speech",
+    "version": "1.0.0-beta00",
+    "type": "grpc",
+    "description": "Recommended Google client library to access the Google Cloud Text-to-Speech API, ynthesizes natural-sounding speech by applying powerful neural network models.",
```
Comment (chrisdunelm) [inline apis/apis.json]: *s*ynthesizes :)

File: apis/Google.Cloud.TextToSpeech.V1/Google.Cloud.TextToSpeech.V1.Snippets/TextToSpeechClientSnippets.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            {
+                LanguageCode = "en-US",
+                SsmlGender = SsmlVoiceGender.Female
+            };
+            // The audio configuration determines the output format and speaking rate.
+            AudioConfig audioConfig = new AudioConfig
+            {
+                AudioEncoding = AudioEncoding.Mp3
+            };
+            var response = client.SynthesizeSpeech(input, voiceSelection, audioConfig);
```
Comment (chrisdunelm) [inline apis/Google.Cloud.TextToSpeech.V1/Google.Cloud.TextToSpeech.V1.Snippets/TextToSpeechClientSnippets.cs]: Don't use `var`

File: apis/Google.Cloud.TextToSpeech.V1/Google.Cloud.TextToSpeech.V1.Snippets/TextToSpeechClientSnippets.cs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+        [Fact]
+        public void ListVoices()
+        {
+            // Snippet: ListVoices(string,CallSettings)
+            TextToSpeechClient client = TextToSpeechClient.Create();
+            ListVoicesResponse response = client.ListVoices("en");
+            foreach (Voice voice in response.Voices)
+            {
+                Console.WriteLine($"{voice.Name}: ({voice.SsmlGender}). Language codes: {string.Join(", ", voice.LanguageCodes)}");
```
Comment (chrisdunelm) [inline apis/Google.Cloud.TextToSpeech.V1/Google.Cloud.TextToSpeech.V1.Snippets/TextToSpeechClientSnippets.cs]: nit: Having output of `:` followed by something in brackets, followed by a `.` seems a little odd?

File: apis/Google.Cloud.TextToSpeech.V1/Google.Cloud.TextToSpeech.V1.Snippets/TextToSpeechClientSnippets.cs [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+            };
+            // The audio configuration determines the output format and speaking rate.
+            AudioConfig audioConfig = new AudioConfig
+            {
+                AudioEncoding = AudioEncoding.Mp3
+            };
+            var response = client.SynthesizeSpeech(input, voiceSelection, audioConfig);
+            using (Stream output = File.Create("sample.mp3"))
+            {
+                response.AudioContent.WriteTo(output);
```
Comment (chrisdunelm) [inline apis/Google.Cloud.TextToSpeech.V1/Google.Cloud.TextToSpeech.V1.Snippets/TextToSpeechClientSnippets.cs]: A comment stating that `AudioContent` is a `ByteString` may be useful.
Or split this into two lines to make this clear in the code.

**Comment by jskeet** [comment]
Thanks for the feedback - will address all and merge.

**Pull request merged**